### PR TITLE
Add PersonLog class (To hold it in RAM) and LogFile class (for storing into a disk)

### LINKED ISSUE #110: Add Person Log Feature

### PR DESCRIPTION
**Pull request by RainyCodeWizard**
PersonLog class under person package- uses singleton design. Only PersonLog class is needed to access the Person logs. 
HashMap is used to store the data.
If need any more methods please tell me thanks.

LogFile class under storage package- Handles storing and retrieving of data into and from a text file. Used object serialization to write objects into a textfile directly.

### CODE REVIEW

File: src/main/java/seedu/duke/Duke.java [hunk 1]
```diff
@@ -87,10 +92,12 @@ private void runUntilExit() {
                 storage.save(trackingList);
                 ui.printReaction(commandOutput);
             } catch (PersonNotFoundException pnfe) {
-                System.out.println("Person not found!");
+                //System.out.println("Person not found!");
+                ui.notifyErrorToUser(pnfe);
```
Comment (sarzorwyn) [inline src/main/java/seedu/duke/Duke.java:96]: it's better to not leave commented code if you are unlikely to use it again

File: src/test/java/seedu/duke/person/PersonLogTest.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        try {
+            personLog.clearAllPersons();
+            personLog.addPerson(ALICE);
+            personLog.addPerson(BOB);
+            personLog.addPerson(JOHN);
+            personLog.addPerson(MARY);
+            personLog.addPerson(JACK_NO_PHONE);
+            personLog.saveAllPersons();
+            personLog.loadAllPersons();
+        } catch (StorageOperationException e) {
```
Comment (sarzorwyn) [inline src/test/java/seedu/duke/person/PersonLogTest.java:31]: Is it possible to separate the save all persons and load all persons test to 2 separate tests?

File: src/main/java/seedu/duke/Duke.java (hunk 1 above)
Comment (iamakilahamed) [reply src/main/java/seedu/duke/Duke.java:96]: Yes I agree too. Likewise for line 70 and 99 in Duke class

File: src/main/java/seedu/duke/person/PersonLog.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    private HashMap<Id, Person> personLog;
+    private LogFile logFile;
+
+    /**
+     * Using Singleton design pattern.
+     */
+    private PersonLog() {
+        personLog = new HashMap<>();
+        logFile = new LogFile();
+    }
```
Comment (iamakilahamed) [inline src/main/java/seedu/duke/person/PersonLog.java:20]: Good use of Singleton design pattern

File: src/main/java/seedu/duke/Duke.java (hunk 1 above)
Comment (RainyCodeWizard) [reply src/main/java/seedu/duke/Duke.java:96]: I wanted to ask you guys first before I deleted the line. On whether I should keep the previous one or current one. 
If if looks ok I will delete the comments in the a new PR

File: src/test/java/seedu/duke/person/PersonLogTest.java (hunk 2 above)
Comment (RainyCodeWizard) [reply src/test/java/seedu/duke/person/PersonLogTest.java:31]: I tried to do it but had synchronisation errors, I will try again later on!

**Comment by iamakilahamed** [comment]
Looks good to merge but @hussain1998 remember to address these problem in the next PR.

**Pull request merged**